test/run-tests

```diff
@@ -723,24 +779,205 @@ class Task:
                             print("FAILURE")
                             return False
 
-                    if not test_passed:
-                        run_status = False
-                        with open(ansible_log, "r") as ansible_file:
-                            for line in ansible_file:
-                                if runqemu.INVENTORY_FAIL_MSG in line:
-                                    print("ERROR: Inventory not properly set up")
-                                    run_status = None
-
-                        print("FAILURE")
-                        logging.debug(f"test failed for {artifactsdir}")
-                    else:
-                        print("SUCCESS")
+                    run_status = self._print_test_result(
+                        test_passed, ansible_log, artifactsdir
+                    )
                     if run_status is None or (
                         not run_status and not args.run_all_tests
                     ):
                         if need_collections_paths and not args.keep_results:
                             cleanup_collection_tempdirs(remove_script=False)
                         return run_status
+
+                if args.backend == KVM_BACKEND:
+                    test_passed = False
+
+                    with redirect_output(output_log, mode="a"):
+                        # Use the qcow2 inventory from standard-test-roles, which
+                        # boots a transient VM and runs the playbook against that.
+                        # Create a fresh instance for each test playbook. However,
+                        # we do need to run the setup (if it exists) in the same
+                        # invocation of ansible-playbook, so that that it applies
+                        # to the same VM as the test playbook.
+                        runqemu_kwargs = dict(
+                            ansible_args=None,
+                            collection_path=collection_dest_path,
+                            pretty=False,
+                            profile=False,
+                            sourcedir=sourcedir,
+                            artifacts=artifactsdir,
+                            setup_yml=[setup_snapshot]
+                            if setup_snapshot != None
```

Comment:
this should be `if setup_snapshot is not None`

Reply:
Oops. Thanks. :p